paddle/fluid/imperative/layer.h

```diff
@@ -182,6 +184,10 @@ class VarBase {
 
   std::string GradVarName() { return framework::GradVarName(Name()); }
 
+  void SetGraphIsFree(bool free) { graph_is_free_ = free; }
+
+  const bool& GraphIsFree() const { return graph_is_free_; }
```

Comment:
`IsFreed` maybe better?

Reply:
Ok

Reply:
I mean `SetGraphIsFreed`, free is not mean status

Reply:
any way, not very importent

Reply:
Get it. Done